src/main/java/org/folio/marc4ld/service/marc2ld/field/FieldMapperImpl.java

```diff
@@ -145,7 +147,10 @@ private void setLabel(Resource resource, Map<String, List<String>> properties, S
     resource.setLabel(
       ofNullable(labelProperty)
         .flatMap(lp -> ofNullable(properties.get(PropertyDictionary.valueOf(lp).getValue())).map(vs -> join(SPACE, vs)))
-        .orElseGet(() -> UUID.randomUUID().toString())
+        .orElseGet(() -> {
+          log.warn("Setting random label to resource of type {}", resource.getTypes());
```

Comment:
Let us add `properties` and `labelProperty ` to this log statement. This way, we will know why we use uuid for label

Reply:
done